rasa/shared/nlu/training_data/features.py

```diff
@@ -115,3 +117,18 @@ def __eq__(self, other: Any) -> bool:
             and other.attribute == self.attribute
             and other.features == self.features
         )
+
+    def fingerprint(self) -> Text:
+        """Calculate a stable string fingerprint for the features."""
+        if isinstance(self.features, np.ndarray):
+            features_as_text = self.features.tostring()
+        else:
+            # sparse features, need to increase printed values
+            current_maxprint = self.features.maxprint
+            # set maxprint to number of values in sparse matrix
+            self.features.maxprint = self.features.nnz
+            features_as_text = str(self.features)
+            self.features.maxprint = current_maxprint
+        return rasa.shared.utils.io.deep_container_fingerprint(
+            [self.type, self.origin, self.attribute, features_as_text]
+        )
```

Comment:
I know it's not great to manipulate the object in place - but given that this is just temporary and only a variable that controls the display of the instance, I thought it was better than transforming the sparse matrix to dense to use `.tostring()`